x-pack/plugins/endpoint/server/routes/alerts/list/lib.ts

```diff
@@ -3,22 +3,69 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { SearchResponse } from 'elasticsearch';
 import { KibanaRequest } from 'kibana/server';
 import {
   JsonObject,
   fromKueryExpression,
   toElasticsearchQuery,
-} from '../../../../../../src/plugins/data/common/es_query/kuery';
-import { EndpointAppConstants } from '../../../common/types';
+} from '../../../../../../../src/plugins/data/common/es_query/kuery';
 import {
-  EndpointAppContext,
-  AlertRequestParams,
-  AlertRequestData,
-  AlertRequest,
-  AlertRequestBody,
-} from '../../types';
-
-function buildQuery(reqData: AlertRequestData): JsonObject {
+  EndpointAppConstants,
+  AlertData,
+  AlertDataWrapper,
+  AlertHits,
+  AlertResultList,
+  ESTotal,
+} from '../../../../common/types';
+import { EndpointAppContext } from '../../../types';
+import {
+  AlertListRequestParams,
+  AlertListRequestData,
+  AlertListRequest,
+  AlertListRequestBody,
+} from './types';
+
+export const getRequestData = async (
+  request: KibanaRequest<unknown, AlertListRequestParams, AlertListRequestParams>,
+  endpointAppContext: EndpointAppContext
+): Promise<AlertListRequestData> => {
+  const config = await endpointAppContext.config();
+  const reqData = {} as AlertListRequestData;
+
+  if (request?.route?.method === 'get') {
+    reqData.pageIndex = request.query?.page_index;
+    reqData.pageSize = request.query?.page_size || config.alertResultListDefaultPageSize;
+    reqData.filters = request.query?.filters || config.alertResultListDefaultFilters;
+    reqData.query = request.query?.query || config.alertResultListDefaultQuery;
+    reqData.sort = request.query?.sort || config.alertResultListDefaultSort;
+    reqData.order = request.query?.order || config.alertResultListDefaultOrder;
+    reqData.searchAfter = request.query?.after;
+    reqData.searchBefore = request.query?.before;
+  } else {
+    reqData.pageIndex = request.body?.page_index;
+    reqData.pageSize = request.body?.page_size || config.alertResultListDefaultPageSize;
+    reqData.filters = request.body?.filters || config.alertResultListDefaultFilters;
+    reqData.query = request.body?.query || config.alertResultListDefaultQuery;
+    reqData.sort = request.body?.sort || config.alertResultListDefaultSort;
+    reqData.order = request.body?.order || config.alertResultListDefaultOrder;
+    reqData.searchAfter = request.body?.after;
+    reqData.searchBefore = request.body?.before;
+  }
+
+  if (reqData.searchAfter === undefined && reqData.searchBefore === undefined) {
+    // simple pagination
+    if (reqData.pageIndex === undefined) {
+      reqData.pageIndex = config.alertResultListDefaultFirstPageIndex;
+    }
+
+    reqData.fromIndex = reqData.pageIndex * reqData.pageSize;
+  }
```

Comment:
It may be hard to deduce what will be the expected output from this function, likely because of the branching. It also opens the door for more branching in the future. It may also be hard to test. I also cannot tell the difference between the get branch and else branch, it appears similar.  Can we simplify?

Reply:
This will be greatly simplified after removing the POST handler. stay tuned.